src/cmd/services/m3coordinator/ingest/write.go

```diff
@@ -94,18 +97,22 @@ type downsamplerAndWriter struct {
 	store       storage.Storage
 	downsampler downsample.Downsampler
 	workerPool  xsync.PooledWorkerPool
+
+	metricsDropped tally.Counter
 }
 
 // NewDownsamplerAndWriter creates a new downsampler and writer.
 func NewDownsamplerAndWriter(
+	scope tally.Scope,
```

Comment:
Instead of passing the scope directly to downsampler and writer, can we pass the `instrument.Options` instead so we get the logger too (and can use the logger later if required)?

Reply:
Also we typically pass the `instrumentOpts instrument.Options` as the last param to each constructor (since instrumentation is least deterministic of how the thing we just constructed will perform).